### PR TITLE
fix: zero slot tasks k8s using wrong image and exposing all GPUs [DET-7808]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Zero slot tasks would use the GPU image before this fix. 

In addition zero slot tasks would end up getting exposed all GPUs (not sure if all in the cluster or on the node but still a problem either way).

## Test Plan

Create a EKS cluster with a GPU

then run and verify that we can see the GPUs (and the image used has ```-gpu-``` in the name)
```
det cmd run --config="resources.slots=1" nvidia-smi
```

now run a zero slot command and verify we do not see any GPUs (should get command not found with image with ```-cpu-``` in the name)
```
det cmd run --config="resources.slots=0" nvidia-smi
```

Then start a notebook verify the image has ```-cpu-``` in the name
```
det notebook start --config="resources.slots=0"
```

Then run the following snippet in a notebook and verify that the output is ```0``` 
```python
import torch
torch.cuda.device_count()
```

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by gh-determined-ai — APPROVED** [review]

**Review by ioga — APPROVED** [review]
awesome
